packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -162,20 +178,31 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     if (_state != _DragState.accepted) {
       _state = _DragState.accepted;
       final Offset delta = _pendingDragOffset;
+      Offset updateDelta = delta;
       final Duration timestamp = _lastPendingEventTimestamp;
+      Offset startPosition;
+      switch (dragStartBehavior) {
+        case DragStartBehavior.start:
+          startPosition = _initialPosition + delta;
+          updateDelta = Offset.zero;
+          break;
+        case DragStartBehavior.down:
+          startPosition = _initialPosition;
+          break;
+      }
       _pendingDragOffset = Offset.zero;
       _lastPendingEventTimestamp = null;
       if (onStart != null) {
         invokeCallback<void>('onStart', () => onStart(new DragStartDetails(
           sourceTimeStamp: timestamp,
-          globalPosition: _initialPosition,
+          globalPosition: startPosition,
         )));
       }
-      if (delta != Offset.zero && onUpdate != null) {
+      if (updateDelta != Offset.zero && onUpdate != null) {
         invokeCallback<void>('onUpdate', () => onUpdate(new DragUpdateDetails(
           sourceTimeStamp: timestamp,
-          delta: _getDeltaForDetails(delta),
-          primaryDelta: _getPrimaryValueFromOffset(delta),
+          delta: _getDeltaForDetails(updateDelta),
+          primaryDelta: _getPrimaryValueFromOffset(updateDelta),
           globalPosition: _initialPosition,
```

Comment:
is it right that this sends the _initialPosition and not the position sent in onStart?

Reply:
Updated. But in this line _initialPosition is the same as startPosition.